frontend/src/heat_map/democracy_viz.js

```diff
@@ -0,0 +1,351 @@
+        const scale = 500;
+        const projection = d3.geoMercator()
+            .center([5, 15])
+            .scale(scale)
+            .translate([scale/2, scale/2]);
+
+        const geoGenerator = d3.geoPath().projection(projection);
+
+        const map_data = [];
+        for (const feature of geo_json.features) {
+            const svg_path = geoGenerator(feature.geometry);
+            const name = feature.properties.name;
+            const postal = feature.properties.postal;
+            const iso = feature.properties.iso_a3; // Not sure if this is the correct ISO code
+            map_data.push({svg_path, name, postal, iso});
+        }
+        return map_data;
+    }
+
+    // handle_country_mouseover(country) {
+    //     this.setState({
+    //         mouseover_country: country.name,
+    //     })
+    // }
+
+    getCountryData(countryCode) {
+        for (const countryData of this.props.democracyData) {
+            if (countryCode === countryData["country_text_id"]) {
+                return countryData;
+            }
+        }
+    }
+
+    render() {
+        if (!this.state.map_data) {
+            return (<div>Loading!</div>);
+        }
+        const colorScale = d3.scaleLinear()
+            .domain([scoreTypeRanges[this.props.scoreType]["min"],
+                scoreTypeRanges[this.props.scoreType]["max"]])
+            .range(['red', 'blue']);
+        return (
+            <>
+                {/*<div>{this.state.mouseover_country}</div>*/}
+
+                <svg
+                    height="1000"
+                    width="1000"
+                    id="content"
+                >
+                    {this.state.map_data.map((country, i) => {
+                        const countryData = this.getCountryData(country.iso);
+                        // Using ternary because some country data doesn't exist
+                        const countryScores = countryData
+                            ? countryData["democracy_scores"][this.props.year]
+                            : null;
+                        const countryScore = countryScores
+                            ? countryScores[this.props.scoreType]
+                            : "";
+                        const color = countryScore !== ""
+                            ? colorScale(countryScore)
+                            : "grey";
+                        return (
+                            <MapPath
+                                key={i}
+                                path={country.svg_path}
+                                id={country.postal}
+                                fill={color}
+                            />
+                        )
+                    })}
+                </svg>
+            </>
+        )
+    }
+}
+DemocracyMap.propTypes = {
+    democracyData: PropTypes.array,
+    scoreType: PropTypes.string,
+    year: PropTypes.string,
+};
+
+export class MapPath extends React.Component {
```

Comment:
This looks like copy/paste from the map quiz -- can you factor out? We'll need this all over the place. Please coordinate with the map quiz team @meesuekim @maoivy @bengao01 to make sure you're not making any contradictory changes to this component.